explore/realspace.py

```diff
@@ -588,7 +588,7 @@ def spin_weights(in_spin, out_spin):
     return weight
 
 def orth(A, b_hat): # A = 3 x n, and b_hat unit vector
- return A - np.sum(A*b_hat[:, None], axis=0)[None, :]*b_hat[:, None]    
+    return A - np.sum(A*b_hat[:, None], axis=0)[None, :]*b_hat[:, None]
```

Comment:
The following goes a beyond the whitespace fixing you did, but:

1) The [:,None] syntax is a bit opaque, maybe b_hat.reshape((3,1))
2) As its used twice, consider declaring it as a variable
3) Consider a more descriptive name and/or comment saying what it does - it's the vector rejection for a an array of vectors right?
4) As the standard formula for this is x - (x dot b_hat) b_hat, maybe code it in terms of np.dot instead.

Reply:
More generally it would be `b_hat.reshape(-1, 1)`. The form `b_hat[:, None]` is pretty common, though it does something [slightly different](https://stackoverflow.com/questions/28385666/numpy-use-reshape-or-newaxis-to-add-dimensions/28385957#28385957). I find the `b_hat[:, None]` form to more readable in expressions and it is ever so slightly faster in my tests, so I will continue using it where appropriate.

Agreed on replacing sum. Given A is 3 x n and b is 3 x 1, then we want `A - (b (A' b)')' = A - b b' A`, which is conveniently expressed as `A - np.outer(b_hat)@A`. This also happens to be faster than `A - b (b' A)`.

I will change it.

Reply:
Nobody chooses python because its fast

Reply:
A - np.outer(b, b)@A is also how it was defined in geni.py. I think to remember there we had a discussion about different ways to construct the orthogonal vector fast and supported by numba.